xbmc/pvr/recordings/PVRRecording.cpp

```diff
@@ -481,3 +481,18 @@ int CPVRRecording::ClientID(void) const
 {
   return m_iClientId;
 }
+
+std::string CPVRRecording::SetGenre(int iGenreType, int iGenreSubType, const char* strGenre)
+{
+  if ((iGenreType == EPG_GENRE_USE_STRING) && (strGenre != NULL) && (strlen(strGenre) > 0))
+  {
+    /* Type and sub type are not given. No EPG color coding possible
+    * Use the provided genre description as backup. */
+    return std::string(strGenre);
+  }
+  else
+  {
+    /* Determine the genre description from the type and subtype IDs */
+    return CEpg::ConvertGenreIdToString(iGenreType, iGenreSubType);
+  }
+}
```

Comment:
well, as this is a member function called "SetGenre" it actually should do that. please change return type to void and set m_genre directly here.